src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts

```diff
@@ -227,8 +275,35 @@ const verifyV3 = async (
 
   const revokedOnAny = revocationStatus.revoked;
 
-  const status = issuedOnAll && !revokedOnAny;
+  if (ValidDidSignedIssuanceStatus.guard(verificationResult) && ValidRevocationStatus.guard(revocationStatus)) {
```

Comment:
Quite weird to see so many guards here, was thinking if it makes sense to have one 

```
if(isValidStatus(verificationResult)){
  return valid stuffs
}else{
  return invalid stuffs (only one more guard is required here to discern between sign and revocation instead of the nested ternary?)
}
```

Reply:
Or the pattern used in `src/verifiers/documentStatus/documentStore/ethereumDocumentStoreStatus.ts` for v2 is also pretty nice. 

if(revoked) return revoked
if(not signed) return not signed
return valid

Reply:
```ts
const issuance = await (await Promise.all(signatureVerificationDeferred)).map(transformToDidSignedIssuanceStatus);
  const notIssued = issuance.find(InvalidDidSignedIssuanceStatus.guard);
  const revoked = revocationStatuses.find(InvalidRevocationStatus.guard);

  const data = {
    issuedOnAll: !notIssued,
    revokedOnAny: !!revoked,
    details: {
      issuance,
      revocation: revocationStatuses,
    },
  };

  if (ValidDidSignedDataV2.guard(data)) {
    return {
      name,
      type,
      data,
      status: "VALID",
    };
  }

  let reason: Reason | undefined;

  if (InvalidDidSignedIssuanceStatus.guard(notIssued)) {
    reason = notIssued.reason;
  }
  if (InvalidRevocationStatus.guard(revoked)) {
    reason = revoked.reason;
  }
  if (!reason) {
    throw new CodedError(
      "Unable to retrieve the reason of the failure",
      OpenAttestationDidSignedDocumentStatusCode.UNEXPECTED_ERROR,
      "UNEXPECTED_ERROR"
    );
  }
  return {
    name,
    type,
    data,
    status: "INVALID",
    reason,
  };
```

Is this ok ?

- no more double guard
- the reason return will be issuance if issuance failed, or revocation, if revoked? (Is it what you meant by like document store ?)

Reply:
lgtm!

Reply:
done